### PR TITLE
Keep the old behaviour with auto generated elasticsearch _id (draft)

### PR DESCRIPTION
**Pull request by vroyer**
If the key is null or "" (the previous behaviour of the ES sink with the provided [extractKeyValue](https://github.com/vroyer/pulsar/blob/branch-2.6/pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/ElasticSearchSink.java#L97)),  the doc is indexed with an auto generated random _id. So the default configuration is now keyIgnore=true, and nullValueAction=IGNORE.

Also add tests for null values.

### CODE REVIEW

File: pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/ElasticSearchSink.java [hunk 1]
```diff
@@ -156,36 +158,33 @@ ElasticSearchClient getElasticsearchClient() {
...
-                        List<String> pkFields = Arrays.asList(elasticSearchConfig.getPrimaryFields().split(","));
-                        id = stringifyKey(jsonNode, pkFields);
-                    } catch (JsonProcessingException e) {
-                        log.error("Failed to read JSON", e);
-                    }
+            if (doc != null && !Strings.isNullOrEmpty(elasticSearchConfig.getPrimaryFields())) {
+                try {
+                    // extract the PK from the JSON document
+                    JsonNode jsonNode = objectMapper.readTree(doc);
+                    List<String> pkFields = Arrays.asList(elasticSearchConfig.getPrimaryFields().split(","));
```
Comment (eolivelli) [inline pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/ElasticSearchSink.java:183]: we can precompute "elasticSearchConfig.getPrimaryFields().split(",")"

this way we can save a few cycles and memory allocations on this hot path

**Review by eolivelli — CHANGES_REQUESTED** [review]
very good.
I left a couple of comments
can you please take a look ?

File: pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/ElasticSearchSink.java (hunk 1 above)
Comment (vroyer) [reply pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/ElasticSearchSink.java:183]: Ok, fixed

File: pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/ElasticSearchSink.java [hunk 2]
```diff
@@ -156,36 +158,33 @@ ElasticSearchClient getElasticsearchClient() {
...
 
             String doc = null;
             if (value != null) {
                 if (valueSchema != null) {
                     doc = stringifyValue(valueSchema, value);
                 } else {
-                    doc = value.toString();
+                    if (value instanceof byte[]) {
+                        // for BWC with the ES-Sink
+                        doc = new String( (byte[]) value);
```
[the remark this answers is not in the document]
Comment (vroyer) [reply pulsar-io/elastic-search/src/main/java/org/apache/pulsar/io/elasticsearch/ElasticSearchSink.java]: This is not good because value is a GenericObject, not byte[]. So we can fix this to be BWC without adding the schemaAware config switch. See the ElasticSearchBWCTests.

**Review by eolivelli — APPROVED** [review]
LGTM
